src/docs/admin/references/NotificationServiceReference.adoc

```diff
@@ -295,3 +295,204 @@ and if they have a value depending on the event origin.
+image::create-app-button.png[align="center"]
+ 2. Provide a name for the App. Here we use the name NotificationService.
+ 3. Leave the default Supported account type in single tenant
+ 4. Finally, click on the "Register" button
+image::microsoft-entra-register-app.png[align="center"]
+
+====== Copy the tenantId and clientID
+
+- If you just created an App, you land on the App overview.
+- If you use an existing App, click on "All applications" and click on your app
+
+Copy and save the tenantId and clientId
+
+image::microsoft-app-tenant-client-id.png[align="center"]
+
+====== Give the app the necessary API permissions
+
+The App needs to have the authorisation to Read Users and Send messages to channels as shown below
+
+image::microsoft-app-api-permission.png[align="center"]
+
+To add these permissions:
+
+- From the App overview, click on the API permission left menu
+- Click on the "Add Permission button"
+- Click on "Microsoft Graph"
+
+image::microsoft-app-graph-permission.png[align="center"]
+
+- Click on "Delegated Permission" (because the authorisation is delegated to users, not the app itself)
+- In the search bar, enter ChannelMessage.Send, select the permission then click on "Add permission"
+
+image::microsoft-add-permission.png[align="center"]
+
+- Do the same for the permission User.Read
+- Click on "Grant admin consent for ..."
+
+image::microsoft-app-grant-admin-consent.png[align="center"]
+
+====== Create a secret
+
+To authenticate itself, the app uses a secret. It must be created on this portal.
+
+WARNING: After the secret is created, it can be read and copied once. If lost, it is definitely lost and a new secret must be created, encoded, and replaced in the Third-Party script.
+
+- If not already there, got to "App registration" and select the App
+- Click on the "Certificates & secrets" in the left menu
+- Click on "New client secret"
+- Provide a secret description and select an expiry date. +
+
+A new secret will need to be created and provided to the script after this date.
+
+IMPORTANT: Copy the secret value as it will be hidden right after. It will be encrypted in the next step.
+
+image::microsoft-copy-secret.png[align="center"]
+
+====== Encrypt the secret
+
+The last step is to encrypt the secret as we don't want to provide it in plain.
+We will also do this for the Microsoft 365 user password.
+
+An encryption tool is provided with ProActive and located in
+----
+PROACTIVE_INSTALL_FOLDER/tools
+----
+
+To encrypt the data:
+- Open a console an navigate to the tools folder of the ProActive installation
+- Execute the encrypt script with the -d option and the data as argument.
+
+Example:
+----
+encrypt -d J8o8Q~q1T54ExSZ262mjBc-zOifI3CfFwuS0GN5srcZN
+----
+
+The output is en encrypt String enclosed by the characters ENC()
+
+Output example:
+----
+ENC(RLqBfho9KuD561d51d8zd62frDF511F5FA51650hPKJGo2p+8mWodQ9C3rF5OcdVg8ea1JrGg=)
+----
+
+You will need the full output, including the enclosing ENC().
```

Comment:
"Ensure the ENC() output is copied accurately, including the parentheses, as it is essential for the script to function properly."

Reply:
OK